pkg/pillar/cmd/zedagent/handlemetrics.go

```diff
@@ -1018,23 +1018,25 @@ func PublishDeviceInfoToZedCloud(ctx *zedagentContext) {
 	ReportDeviceInfo.HSMInfo, _ = tpmmgr.FetchTpmHwInfo()
 
 	//Operational information about Data Security At Rest
-	ReportDataSecAtRestInfo := new(info.DataSecAtRest)
-	var vaultList []vaultmgr.VaultStatus
-	ReportDataSecAtRestInfo.Status, ReportDataSecAtRestInfo.Info, vaultList =
-		vaultmgr.GetOperInfo()
-	ReportDataSecAtRestInfo.VaultList = make([]*info.VaultInfo, 0)
-	for _, vault := range vaultList {
-		vaultInfo := new(info.VaultInfo)
-		vaultInfo.Name = vault.Name
-		vaultInfo.Status = vault.Status
-		if !vault.ErrorTime.IsZero() {
-			vaultInfo.VaultErr = new(info.ErrorInfo)
-			vaultInfo.VaultErr.Description = vault.Error
-			vaultInfo.VaultErr.Timestamp, _ = ptypes.TimestampProto(vault.ErrorTime)
-		}
-		ReportDataSecAtRestInfo.VaultList = append(ReportDataSecAtRestInfo.VaultList, vaultInfo)
-	}
-	ReportDeviceInfo.DataSecAtRestInfo = ReportDataSecAtRestInfo
+	getDataSecAtRestInfo := func() *info.DataSecAtRest {
```

Comment:
Instead of inlining the function definition, can you make a normal function? I suspect it just needs ctx.subVaulStatsu (or ctx) as an argument.

Reply:
Addressed in the latest patch.